c10/core/SymInt.h

```diff
@@ -209,15 +209,10 @@ class C10_API SymInt {
       return c10::make_optional(data_);
     }
     auto* node = toSymNodeImplUnowned();
-    int64_t c = node->large_negative_int();
-    if (c != 0) {
-      return c10::make_optional(c);
+    if (auto c = node->constant_int()) {
+      return c;
     }
-    c10::optional<int64_t> d = node->maybe_as_int();
-    if (d.has_value()) {
-      return d;
-    }
-    return c10::nullopt;
+    return node->maybe_as_int();
```

Comment:
This is something of a preexisting problem, but I wonder why I didn't just have `maybe_as_int` take care of everything...

Reply:
Filing an issue to do this later https://github.com/pytorch/pytorch/issues/107873